pkg/apis/networking/validation/validation.go

```diff
@@ -259,45 +293,73 @@ func validateIngressRules(ingressRules []networking.IngressRule, fldPath *field.
+		allErrs = append(allErrs, validateHTTPIngressPath(&path, fldPath.Child("paths").Index(i), opts)...)
+	}
+	return allErrs
+}
+
+func validateHTTPIngressPath(path *networking.HTTPIngressPath, fldPath *field.Path, opts IngressValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if path.PathType != nil {
+		if !supportedPathTypes.Has(string(*path.PathType)) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("pathType"), path.PathType, supportedPathTypes.List()))
+		}
+
+		switch *path.PathType {
+		case networking.PathTypeExact, networking.PathTypePrefix:
+			if !strings.HasPrefix(path.Path, "/") {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), path.Path, "must be an absolute path"))
 			}
-			// TODO: More draconian path regex validation.
-			// Path must be a valid regex. This is the basic requirement.
-			// In addition to this any characters not allowed in a path per
-			// RFC 3986 section-3.3 cannot appear as a literal in the regex.
-			// Consider the example: http://host/valid?#bar, everything after
-			// the last '/' is a valid regex that matches valid#bar, which
-			// isn't a valid path, because the path terminates at the first ?
-			// or #. A more sophisticated form of validation would detect that
-			// the user is confusing url regexes with path regexes.
-			_, err := regexp.CompilePOSIX(rule.Path)
-			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("paths").Index(i).Child("path"), rule.Path, "must be a valid regex"))
+			if len(path.Path) > 0 && strings.Contains(path.Path, "//") {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), path.Path, "must not contain repeating '/' characters"))
 			}
 		}
-		allErrs = append(allErrs, validateIngressBackend(&rule.Backend, fldPath.Child("backend"))...)
 	}
+
+	if path.PathType == nil || *path.PathType == networking.PathTypeImplementationSpecific {
+		if len(path.Path) > 0 {
+				if !strings.HasPrefix(path.Path, "/") {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), path.Path, "must be an absolute path"))
+				}
+			    if opts.requireRegexPath {
+					// TODO: More draconian path regex validation. Path must be a
+					// valid regex. This is the basic requirement. In addition to
+					// this any characters not allowed in a path per RFC 3986
+					// section-3.3 cannot appear as a literal in the regex. Consider
+					// the example: http://host/valid?#bar, everything after the
+					// last '/' is a valid regex that matches valid#bar, which isn't
+					// a valid path, because the path terminates at the first ? or
+					// #. A more sophisticated form of validation would detect that
+					// the user is confusing url regexes with path regexes.
+					_, err := regexp.CompilePOSIX(path.Path)
+					if err != nil {
+						allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), path.Path, "must be a valid regex"))
+					}
+				}
+		}
```

Comment:
the `/` prefix check and conditional "valid regex" validation must be done for all pathTypes in 1.18 or we'll allow in data that a 1.17 API server would choke on

Reply:
Updated. To clarify, these are still allowed as part of updates to Ingresses that already have non-regex path types, but otherwise are considered invalid.